airflow/providers/apache/beam/operators/beam.py

```diff
@@ -91,6 +92,13 @@ def __get_dataflow_pipeline_options(
             pipeline_options[job_name_key] = job_name
         if self.dataflow_config.service_account:
             pipeline_options["serviceAccount"] = self.dataflow_config.service_account
+        if self.dataflow_support_impersonation and self.dataflow_config.impersonation_chain:
+            if isinstance(self.dataflow_config.impersonation_chain, list):
+                pipeline_options["impersonateServiceAccount"] = ",".join(
+                    self.dataflow_config.impersonation_chain
+                )
+            else:
+                pipeline_options["impersonateServiceAccount"] = self.dataflow_config.impersonation_chain
```

Comment:
Instead of needing an extra flag, is it possible to always add the `impersonateServiceAccount` value? Or would it error if it’s present when not applicable?

Reply:
The behavior of apache beam can change from version to version, so I'd rather not add extra parameters when we don't need it.

Reply:
For now Apache beam added this option for Java and Python SDK and if we pass this attr to the GO without new flag we will get error in the execution.